loopy/check.py

```diff
@@ -380,8 +381,12 @@ def __init__(self, kernel, domain, insn_id):
         self.domain = domain
```

Comment:
I think this would be less error-prone if we removed the global `domain` here and simply passed the domain down the recursion in the same way you currently do with `context`.

Reply:
Done. (Squeezed the domain and context into one, since that's all what we need)